transport/transport-runtime/src/androidTest/java/com/google/android/datatransport/runtime/TransportRuntimeRule.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.runtime;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public final class TransportRuntimeRule implements TestRule {
+  private TransportRuntimeComponent component;
+
+  TransportRuntimeRule(TransportRuntimeComponent component) {
+    this.component = component;
+  }
+
+  @Override
+  public Statement apply(final Statement base, final Description description) {
+    return new Statement() {
+      @Override
+      public void evaluate() throws Throwable {
+        TransportRuntime.withInstance(
+            component,
+            () -> {
+              try {
+                base.evaluate();
+              } catch (Throwable throwable) {
```

Comment:
I think this catch defeats the purpose of the tests as all tests will pass no matter what. I think the easiest way to fix would be to switch to a `Callable` in `withInstance()` and change its signature to something like:
```java
static <T> T withInstance(Component cmp, Callable<T> callable) throws Throwable
```